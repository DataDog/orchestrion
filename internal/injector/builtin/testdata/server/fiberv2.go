// Unless explicitly stated otherwise all files in this repository are licensed
// under the Apache License Version 2.0.
// This product includes software developed at Datadog (https://www.datadoghq.com/).
// Copyright 2023-present Datadog, Inc.

package main

import (
	"github.com/gofiber/fiber/v2"
	fibertrace "gopkg.in/DataDog/dd-trace-go.v1/contrib/gofiber/fiber.v2"
)

func fiberV2Server() {
	//dd:instrumented
	r := fiber.New()
//line <generated>:1
	//dd:startinstrument
	{
		r.Use(fibertrace.Middleware())
	}
	//dd:endinstrument
//line samples/server/fiberv2.go:15
	r.Use(fibertrace.Middleware())
	r.Get("/ping", func(c *fiber.Ctx) error {
		return c.JSON(map[string]any{
			"message": "pong",
		})
	})
	_ = r.Listen(":8080")
}
