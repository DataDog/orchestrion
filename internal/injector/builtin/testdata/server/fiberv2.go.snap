// Unless explicitly stated otherwise all files in this repository are licensed
// under the Apache License Version 2.0.
// This product includes software developed at Datadog (https://www.datadoghq.com/).
// Copyright 2023-present Datadog, Inc.

package main

import (
	"github.com/gofiber/fiber/v2"
//line <generated>:1
	fiber1 "gopkg.in/DataDog/dd-trace-go.v1/contrib/gofiber/fiber.v2"
)

//line samples/server/fiberv2.go:12
func fiberV2Server() {
	r :=
//line <generated>:1
		func() *fiber.App {
			app :=
//line samples/server/fiberv2.go:13
				fiber.New()
//line <generated>:1
			app.Use(fiber1.Middleware())
			return app
		}()
//line samples/server/fiberv2.go:14
	r.Get("/ping", func(c *fiber.Ctx) error {
		return c.JSON(map[string]any{
			"message": "pong",
		})
	})
	_ = r.Listen(":8080")
}
